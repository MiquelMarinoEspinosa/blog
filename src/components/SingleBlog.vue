<template>
  <div id="single-blog">
    <div v-if="success">
      <h1>{{ blog.title }}</h1>
      <article>{{ blog.content }}</article>
      <p>Author: {{ blog.author }}</p>
      <p>Categories:</p>
      <ul>
        <li v-for="(category, index) in blog.categories" :key="index">{{ category }}</li>
      </ul>
    </div>
    <div v-else>
      <h1>Blog not found</h1>
      <article>ID: {{ id }}</article>
      <p>{{ error }}</p>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      id: this.$route.params.id,
      blog: {},
      success: true,
      error: "",
    };
  },
  created() {
    this.$http
      .get("http://localhost:8081/blog/" + this.id)
      .then((response) => {
        this.blog = response.data;
        this.success = true;
        this.error = "";
      })
      .catch((error) => {
        this.success = false;
        this.error = error;
      });
  },
};
</script>

<style>
#single-blog {
  max-width: 960px;
  margin: 0 auto;
}
</style>
