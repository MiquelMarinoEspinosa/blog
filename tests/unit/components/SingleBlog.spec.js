import { createLocalVue, shallowMount } from "@vue/test-utils";
import axios from "axios";
import SingleBlog from "@/components/SingleBlog";

const localVue = createLocalVue();

jest.mock("axios");

localVue.prototype.$http = axios;

describe("SingleBlog", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("retrieves the single blog", async () => {
    const id = "e2e6d160-15ff-4d30-92ce-2799eb7fe5bd";
    const $route = {
      params: {
        id,
      },
    };
    const data = {
      title: "Blog 1",
      content: "Blog content 1",
      author: "Author 1",
      categories: ["vue", "react", "node"],
    };
    axios.get.mockResolvedValue({ data });
    const singleBlog = shallowMount(SingleBlog, {
      localVue,
      mocks: { $route },
    });
    await localVue.nextTick();
    expect(axios.get).toBeCalledTimes(1);
    expect(axios.get).toBeCalledWith("http://localhost:8081/blog/" + id);
    expect(singleBlog.html()).toMatchSnapshot();
  });

  it("has happened an error and no blog was retreived", async () => {
    const id = "e2e6d160-15ff-4d30-92ce-2799eb7fe5b";
    const $route = {
      params: {
        id,
      },
    };
    axios.get.mockRejectedValue("Something went wrong :(");
    const singleBlog = shallowMount(SingleBlog, {
      localVue,
      mocks: { $route },
    });
    await localVue.nextTick();
    expect(axios.get).toBeCalledTimes(1);
    expect(axios.get).toBeCalledWith("http://localhost:8081/blog/" + id);
    await localVue.nextTick();
    expect(singleBlog.html()).toMatchSnapshot();
  });
});
