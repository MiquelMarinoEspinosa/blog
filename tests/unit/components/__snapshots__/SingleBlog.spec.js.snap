// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleBlog retrieves the single blog 1`] = `
<div id="single-blog">
  <div>
    <h1>Blog 1</h1>
    <article>Blog content 1</article>
    <p>Author: Author 1</p>
    <p>Categories:</p>
    <ul>
      <li>vue</li>
      <li>react</li>
      <li>node</li>
    </ul>
  </div>
</div>
`;

exports[`SingleBlog has happened an error and no blog was retreived 1`] = `
<div id="single-blog">
  <div>
    <h1>Blog not found</h1>
    <article>ID: e2e6d160-15ff-4d30-92ce-2799eb7fe5b</article>
    <p>Something went wrong :(</p>
  </div>
</div>
`;